### ISSUE TITLE
How can Company manager or department manager can assign roles into course

### ISSUE THREAD

**Issue by shail2206** [iomad/iomad#348]
Hi,
I have some doubts ..
1. How can Company Manager or Department Manager assign different roles (except "user role" this is the only value I can see in dropdown) to particular course.
2. "Add and Activity and Resource" display empty box. Is this functionality or bug ? how can Company manager and department Manager manage courses ?
3. what is the use of "Client Course Editor"  Role? Company manager not even able to assign this role (this is default role in IOMAD system). Even after assign this role by moodle's default tool, this role don't have any permission to manage course.

In short .... Please guide me Roles & access control flow in IOMAD and what about course management ? is there any documentation on this ? is yes please share the link.

Thanks

**Comment by turf212** [comment]
If a company manager creates a course then they are given the client course editor role in the course.  This is like the editing teacher role so they can enrol users and create content and more in it.  This would be the only way a company manager could provide a different role to a user.  Due to the way the multi-tennancy works on shared courses you need to be very careful how roles are being asigned as it may allow for a company manager to see users outwith their company.

You should be able to see a number of activities and resources (unless you have turned them off at a site level).  Can you please outline the exact steps you used to find this?

**Comment by shail2206** [comment]
Hi,

I found the solution, actually problem was in default capabilities of "Client Course Editor" Role.
by default this role doesn't have any permission like adding any activity and resource into course, user enrollment ,etc. that's why I am not able to add any activity.
for the time being I rename existing role and create another role with same name with Teacher Role capability; now everything is working fine. User can add resource and able to assign roles inside course.

I have just a question in my mind; why this setting was by default ? after my changes what impact you can foresee ?

![iomad_course_view_page](https://cloud.githubusercontent.com/assets/4610115/15348780/01bf3a6a-1ceb-11e6-9832-d93b6eab6dc8.png)

Please Note: This was untouched IOMAD 2.9 version. not any changes on configuration and coding level.

Please share your view.

Thanks

**Comment by turf212** [comment]
Thanks for the update.  I'll check the permissions for install on this role and make sure these are captured.
